### DISCUSSION TITLE
Issues with editing input texts within a form

### DISCUSSION THREAD

**Discussion by colorfularty** [WarsWorld/WarsWorld#126] [Q&A]
Has anyone else run into an issue where text input fields in web forms are not working properly? I'm making a page for people to set/update their username and tagline using input texts, but it is not displaying the default values I've set for them, nor am I able to edit them when testing the page. If anyone else has encountered this and found a solution, that would really help me out. I've been trying to troubleshoot this for hours.

**Comment by leekahung** [comment]
Hey Arty! How are you writing up the form at the moment? Perhaps we could help you debug some of it from snippets.

**Comment by colorfularty** [comment]
Screenshots of the code, since it won't let me attach or paste the code without screwing up the formatting:
![Screenshot (232)](https://github.com/WarsWorld/WarsWorld/assets/35205060/222a1d85-e72f-44de-b2b4-5c0c97d1ebd5)
![Screenshot (233)](https://github.com/WarsWorld/WarsWorld/assets/35205060/dd3e3a14-dbcb-4ff8-95a0-4d6f396d3231)
![Screenshot (234)](https://github.com/WarsWorld/WarsWorld/assets/35205060/6dcfeb20-87b5-4cfa-adf5-3e9cacc097de)
![Screenshot (235)](https://github.com/WarsWorld/WarsWorld/assets/35205060/081560eb-2f18-4876-82e6-eaee073dc771)
![Screenshot (236)](https://github.com/WarsWorld/WarsWorld/assets/35205060/ef69f6cf-2f72-437c-a443-d7cd39bd7eb5)

**Reply by leekahung** [reply]
Hey Arty, thanks for the screenshots! I'll try my best to answer the errors in the code and any other suggestions. I'll have the screenshot first then leave a comment below. But in short, after looking at the code, it's likely due to the fact the form you're making is not set up properly. I've made some comments that might be useful to you. I've also added comments at the end of the reply which should be a great resource for you. I hope this helps!

---

> ![Screenshot (232)](https://user-images.githubusercontent.com/35205060/238487368-222a1d85-e72f-44de-b2b4-5c0c97d1ebd5.png)

For the first screenshot, I believe you're trying to make a handler function? In that case, I think you should be calling the specific form value using the HTML id attribute attached to an input in the component JSX rather than calling document. The reason why document is not picking up "profileChange" because it's not part of the original DOM.

> ![Screenshot (233)](https://user-images.githubusercontent.com/35205060/238487371-dd3e3a14-dbcb-4ff8-95a0-4d6f396d3231.png)

For the second screenshot, Line 54 is not yet typed, which is why the error is appearing.

As for the second error, JSX uses camel case for their attributes, so `onsubmit` should be `onSubmit`.

Also to note, since we're using Tailwind and SASS, I think you can simply add the class to the HTML element via a local state which could be toggled to turn on and off that class when clicked.

> ![Screenshot (234)](https://user-images.githubusercontent.com/35205060/238487372-6dcfeb20-87b5-4cfa-adf5-3e9cacc097de.png)

For the third screenshot, I've started to notice a lot of repetitive code with the divs. Perhaps you could refactor them into smaller components the select and option elements (see: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select) and calling them using curly braces `{}` and map functions. This way we could reduce the amount of code being repeated in the JSX and avoid potential errors.

Also for the onClick in Line 86, the error came about because you're not using curly braces `{}` for running the handler function in JSX. As for the handler function, I recommend going over React's new documentation (https://react.dev/learn/responding-to-events#adding-event-handlers). It does a really good job showing they should write them in React.

> ![Screenshot (235)](https://user-images.githubusercontent.com/35205060/238487375-081560eb-2f18-4876-82e6-eaee073dc771.png)

This is screenshot has the same repetitive code issue as above, see answer for the third screenshot.

> ![Screenshot (236)](https://user-images.githubusercontent.com/35205060/238487378-ef69f6cf-2f72-437c-a443-d7cd39bd7eb5.png)

This is screenshot has the same repetitive code issue as above, see answer for the third screenshot.

Additionally, Line 179 should be a button element for form submission, not an input element. I'm also guessing that Line 181 is suppose to be a normal button with an event handler to exit the form. A Link for that is not needed.

---

In case you need any additional guidance, I personally recommend the new React documentation. It does a nice job teaching and providing examples of how one should create components in React. I recommend it. With regards to your forms, check out the following link: https://react.dev/reference/react-dom/components/input#controlling-an-input-with-a-state-variable

**Comment by colorfularty** [comment]
Thanks so much for the detailed comments! I'll definitely look into these and try your suggestions!

**Reply by leekahung** [reply]
You're welcome! Feel free to comment here or on Discord if you have any other questions related to the code.

**Reply by colorfularty** [reply]
Well I feel stupid. The reason I couldn't edit the text window is because the text was the exact same color as the background, so I just couldn't see it! XD

Thanks a lot for the help! Is there a specific discord for WarsWorld yet?

**Reply by leekahung** [reply]
We do have a private Discord channel set up for the team. I'll mention @FemboyDeveloper here.

**Comment by FunctionDJ** [comment]
i only now saw the screenshots. we're working with react, not plain HTML where what you wrote would be valid. you can't use strings as functions like in vanilla HTML.

**Reply by colorfularty** [reply]
I guess I don't fully understand what you're saying. I wrote my code similar enough to the other code that i thought it would work. I tested it and it worked for me by using the pnpm dx through docker.
